pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/nonpepperkit/NonPepperKitCreationService.java

```diff
@@ -0,0 +1,276 @@
+     *
+     * @return SimpleResult of the interaction with database
+     */
+    private SimpleResult createKit(@NonNull DDPInstance ddpInstance, @NonNull KitType kitType, JuniperKitRequest kit,
+                                   @NonNull KitRequestSettings kitRequestSettings, @NonNull EasyPostUtil easyPostUtil,
+                                   @NonNull String kitTypeName, ArrayList<KitRequest> orderKits, Connection conn,
+                                   SimpleResult transactionResults) {
+        String juniperKitRequestId;
+        String userId;
+        //checking ddpInstance.isHasRole() to know this is a Juniper Kit
+        if (StringUtils.isNotBlank(kit.getJuniperKitId()) || ddpInstance.isHasRole()) {
+            juniperKitRequestId = JUNIPER_UNDERSCORE + kit.getJuniperKitId();
+            userId = JUNIPER;
+        } else {
+            log.warn("Seems like {} is not configured as a JUNIPER study! ", ddpInstance.getName());
+            transactionResults.resultException = new Exception(ddpInstance.getName() + " study is not configured to use this method.");
+            return transactionResults;
+        }
+
+        String externalOrderNumber = null;
+        if (StringUtils.isNotBlank(kitRequestSettings.getExternalShipper())) {
+            externalOrderNumber = DDPKitRequest.generateExternalOrderNumber();
+        }
+        String errorMessage = "";
+        // collaboratorParticipantId is the id of the participant in GP's view
+        String collaboratorParticipantId = KitRequestShipping
+                .getCollaboratorParticipantId(null, ddpInstance.getDdpInstanceId(), ddpInstance.isMigratedDDP(),
+                        ddpInstance.getCollaboratorIdPrefix(), kit.getJuniperParticipantID(), kit.getJuniperParticipantID(),
+                        kitRequestSettings.getCollaboratorParticipantLengthOverwrite());
+
+        //In case it's needed, subkits handling should be added here
+
+        addJuniperKitRequest(conn, kitTypeName, kitRequestSettings, ddpInstance, kitType.getKitTypeId(), collaboratorParticipantId,
+                errorMessage, easyPostUtil, kit, externalOrderNumber, juniperKitRequestId, null, userId, transactionResults);
+
+        // Not needed now, uncomment later when needed for external shippers
+        //        orderKits.add(kit);
+
+        return transactionResults;
+    }
+
+    private SimpleResult addJuniperKitRequest(Connection conn, String kitTypeName, KitRequestSettings kitRequestSettings,
+                                              DDPInstance ddpInstance,
+                                              int kitTypeId, String collaboratorParticipantId, String errorMessage,
+                                              EasyPostUtil easyPostUtil,
+                                              JuniperKitRequest kit, String externalOrderNumber, String juniperKitRequestId,
+                                              String ddpLabel, String userId, SimpleResult transactionResults) {
```

Comment:
I'm not a fan of lots of parameters -- it makes for poor readability. Is there a way to encapsulate these into a class that is acting as a struct? Or more conventionally, create a kit shipping class that has much of this info as instance variables.